ykrt/src/compile/j2/x64/x64hir_to_asm.rs

```diff
@@ -5777,6 +5782,22 @@ mod test {
 
         // Variable RHS
 
+        // i8
+        codegen_and_test(
+            "
+              %0: i8 = arg [reg]
+              %1: i8 = arg [reg]
+              %2: i8 = lshr %0, %1
+              exit [%0, %2]
+            ",
+            &["
+              ...
+              ; %2: i8 = lshr %0, %1
+              shr r.32.x, cl
```

Comment:
I forgot that this meant a wildcard 32-bit register.. Just curious: why does it have to be 32-bit though?

Reply:
Right. I was thinking if it were 64-bit instead. However, I think that probably causes more register pressure? So maybe we don't want that.

Reply:
The advantage of using 32-bit registers is that, for 32 bit values, we implicitly zero extend them to 64-bits (i.e. we get some zero extension for free), which helps a little in some cases.

Reply:
Cool. Thank you for the explanation.